libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL-fp.cpp

```diff
@@ -0,0 +1,36 @@
+
+#include <WiFiClientSecureBearSSL.h>
+
+namespace BearSSL {
+
+static uint8_t htoi (unsigned char c)
+{
+    return c >= '0' && c <= '9'? c - '0'
+         : c >= 'A' && c <= 'F'? c - 'A' + 10
+         : c >= 'a' && c <= 'f'? c - 'a' + 10
+         : 255;
+}
+
+void WiFiClientSecure::setFingerprint(const String& fingerprint) {
```

Comment:
Does a C string (`"12 34 56 ..."`) silently promote to a String&?  Maybe using a `char *` would be better WRT memory.

Reply:
Yes. The advantage of `const String&` is that it will take a progmem or ram char*. It will be released right after `setFingerprint(fp);` which does not any allocation, so ram won't be holed.

Reply:
I think that in this case a C string promotes to a temp String object, which is then passed in by reference.